source/extensions/common/wasm/wasm.cc

```diff
@@ -106,7 +106,8 @@ Wasm::Wasm(WasmConfig& config, absl::string_view vm_key, const Stats::ScopeShare
 Wasm::Wasm(WasmHandleSharedPtr base_wasm_handle, Event::Dispatcher& dispatcher)
     : WasmBase(base_wasm_handle,
                [&base_wasm_handle]() {
-                 return createWasmVm(base_wasm_handle->wasm()->wasm_vm()->runtime());
+                 return createWasmVm(absl::StrCat("envoy.wasm.runtime.",
```

Comment:
how this fix non-cloneable runtimes?

Reply:
`base_wasm_handle->wasm()->wasm_vm()->runtime()` is part of Proxy-Wasm C++ Host, which returns `"v8"`, etc.

`createWasmVm(name)` is Envoy's interface for Wasm Runtimes, which uses Envoy names `"envoy.wasm.runtime.v8"`, etc.

As such, there was a mismatach when using this constructor (which is only used for non-cloneable runtimes, which we don't have right now), because calling `createWasmVm("v8")` returns `"Failed to create Wasm VM using v8 runtime. Envoy was compiled without support for it"`, since that runtime name is unknown to Envoy.